app/controllers/api/events_controller.rb

```diff
@@ -36,7 +36,10 @@ def process_incoming_event
           was_new = upp.new_record?
           upp.legacy_count[create_params[:workflow]] = create_params[:count]
           if upp.save
-            Project.increment_counter :classifiers_count, upp.project_id if was_new
+            if was_new
+              Project.increment_counter :classifiers_count, upp.project_id
+              upp.user.update( {project_id: create_params[:project_id]} )
```

Comment:
@zwolf the source of the FK violations are due to using the legacy project id her and not the id of the project itself. if you chnange this to `upp.update_column(:project_id, upp.project_id)` it should not violate the FK constraint.